### ISSUE TITLE
citra is broken

### ISSUE THREAD

**Issue by rnhmjoj** [NixOS/nixpkgs#34212]
## Issue description
The citra package is currently failing to build.
```
       return iter != table.end() ? boost::make_optional<const VFP2Matcher<V>&>(*iter) : boost::none;
              ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  compilation terminated due to -Wfatal-errors.
  make[2]: *** [externals/dynarmic/src/CMakeFiles/dynarmic.dir/build.make:111: externals/dynarmic/src/CMakeFiles/dynarmic.dir/frontend/disassembler/disassembler_arm.cpp.o] Error 1
  make[1]: *** [CMakeFiles/Makefile2:219: externals/dynarmic/src/CMakeFiles/dynarmic.dir/all] Error 2
  make: *** [Makefile:141: all] Error 2
```

usual boost breaking change?

### Steps to reproduce
`$ nix run nixpkgs.citra`

## Technical details
I tried to update the sources but fetchgit fails with
```
fatal: '9ab2117c5bac2f376f8da4a4b25e183137bcec0' is not a commit and a branch 'fetchgit' cannot be created from it
Unable to checkout 9ab2117c5bac2f376f8da4a4b25e183137bcec0 from https://github.com/philsquared/Catch.git.
builder for '/nix/store/79fnsp1sh33f3367iljbwsxfnjw3g6ca-citra-743c247.drv' failed with exit code 1
error: build of '/nix/store/79fnsp1sh33f3367iljbwsxfnjw3g6ca-citra-743c247.drv' failed
```
No idea what this means.

---

cc @abbradar

**Comment by rnhmjoj** [comment]
Yes, it's boost 1.66 that broke it but I still can't update the source.

**Comment by dtzWill** [comment]
Source updating/fetching should be fixed with #34232 .

**Comment by rnhmjoj** [comment]
Ah, thank you.
